R/themePubr.R

```diff
@@ -0,0 +1,103 @@
+#' with other themes, \code{graph + themePubrRaw()}.
+#' @rdname themePubr
+#'
+#' @description This function is a modification of 
+#'  \code{\link[ggpubr:theme_pubr]{ggpubr::theme_pubr}} that create a publication ready
+#'  theme.
+#' @param base_size base font size
+#' @param base_family base font family
+#' @param border logical value. Default is FALSE. If TRUE, add panel border.
+#' @param margin logical value. Default is TRUE. If FALSE, reduce plot margin.
+#' @param legend character specifying legend position. Allowed values are one of
+#'  c("top", "bottom", "left", "right", "none"). Default is "top" side position.
+#'  to remove the legend use legend = "none". Legend position can be also
+#'  specified using a numeric vector c(x, y).  In this case it is possible to
+#'  position the legend inside the plotting area. x and y are the coordinates of
+#'  the legend box. Their values should be between 0 and 1. c(0,0) corresponds
+#'  to the "bottom left" and c(1,1) corresponds to the "top right" position. For
+#'  instance use legend = c(0.8, 0.2).
+#' @param x.text.angle Rotation angle of x axis tick labels. Default value is 0.
+#'  Use 90 for vertical text.
+#' @param flip logical. If TRUE, grid lines are added to y axis instead of x
+#'  axis.
+#' @param axisTickLength length of axis ticks.
+#' @param axisTickWidth width of axis ticks.
+#' @export
+themePubr <- function(base_size = getGraphOption("fontsize"),
+                      base_family = getGraphOption("family"),
+                      border = FALSE,
+                      margin = TRUE,
+                      legend = getGraphOption("legend.position"),
+                      x.text.angle = 0,
+                      axisTickLength = getGraphOption("axisTickLength"),
+                      axisTickWidth  = getGraphOption("axisTickWidth")){
+  
+  
+  graph <- graph + themePubrRaw(base_size   = base_size,
+                                base_family = base_family,
+                                border = FALSE,
+                                margin = TRUE,
+                                legend = legend,
+                                x.text.angle = 0,
+                                axisTickLength = axisTickLength,
+                                axisTickWidth  = axisTickWidth)
+  
+  return(graph)
+  
+}
+
+# for manual usage
+#' @rdname themeRpur
+#' @export
+themePubrRaw <- function(base_size = getGraphOption("fontsize"),
+                         base_family = getGraphOption("family"),
+                         border = FALSE,
+                         margin = TRUE,
+                         legend = getGraphOption("legend.position"),
+                         x.text.angle = 0,
+                         axisTickLength = getGraphOption("axisTickLength"),
+                         axisTickWidth  = getGraphOption("axisTickWidth")){
+  
+  half_line <- base_size/2
+  
+  if (!is.numeric(legend)) 
+    legend <- match.arg(legend)
```

Comment:
This doesn't do anything unless the legend argument actually contains default values. Rather, it'll actually make it impossible to use strings that are unequal to `getGraphOption("legend.position")`